.github/actions/link-package/action.yml

```diff
@@ -0,0 +1,47 @@
+name: Build Lumino from source and link to JupyterLab
+description: |
+  Build Lumino from source then link to JupyterLab.
+  Useful to test changes in Lumino against a given build of JupyterLab.
+
+inputs:
+  external_pkg_repo:
+    description: |
+      External package repository to link, for example jupyter/lumino
+      only to be used if you want to link and test a package against JupyterLab.
+      Otherwise leave blank.
+    required: false
+    default: ""
+  external_pkg_ref:
+    description: |
+      External package repository's ref (branch, tag or SHA) to checkout,
+      only to be used if you want to link and test a package against JupyterLab.
+      Otherwise leave blank.
+    required: false
+    default: ""
+
+runs:
+  using: composite
+  steps:
+    - name: Checkout package repo 🛎
+      uses: actions/checkout@v3
+      with:
+        repository: ${{ inputs.external_pkg_repo }}
+        ref: ${{ inputs.external_pkg_ref }}
+        path: external-pkg
+
+    # following https://github.com/jupyterlab/lumino/blob/main/CONTRIBUTING.md#setting-up-a-development-environment
+    - name: Build package from source 🏗
+      run: |
+        yarn
+        yarn build:src
+        conda run --prefix $CONDA_PREFIX jlpm link
+      shell: bash -l {0}
+      working-directory: external-pkg
+
+    - name: Link JupyterLab and package 🔗
+      run: |
+        # todo later on: replace the lumino reference here
+        conda run --prefix $CONDA_PREFIX jlpm link "lumino-top-level"
```

Comment:
Does this work? I'm not sure but I think you actually have to go through each of the (sub)packages in the Lumino monorepo and link them one by one. 

Is that what your "todo" comment is about?

Reply:
This works as is - will make improvements for robustness in a later iteration

Reply:
Yeah, I remember we chatted about this with @steff456, I think, and I may have just been flat out wrong about needing to link every sub-package versus the top-level package.